tests/Test.ADAL.NET.Integration/SovereignCloudsTests.cs

```diff
@@ -61,15 +97,103 @@ public void Initialize()
                 TestConstants.DefaultRedirectUri, platformParameters, UserIdentifier.AnyUser, "instance_aware=true");
 
             // make sure that tenant spesific sovereign Authority returned to the app in AuthenticationResult
-            Assert.AreEqual(sovereignTenantSpesificAuthority, authenticationResult.Authority);
+            Assert.AreEqual(sovereignTenantSpecificAuthority, authenticationResult.Authority);
 
             // make sure that AuthenticationContext Authority was updated
-            Assert.AreEqual(sovereignTenantSpesificAuthority, authenticationContext.Authority);
+            Assert.AreEqual(sovereignTenantSpecificAuthority, authenticationContext.Authority);
 
             // make sure AT was stored in the cache with tenant spesific Sovereign Authority in the key
             Assert.AreEqual(1, authenticationContext.TokenCache.tokenCacheDictionary.Count);
-            Assert.AreEqual(sovereignTenantSpesificAuthority,
+            Assert.AreEqual(sovereignTenantSpecificAuthority,
                 authenticationContext.TokenCache.tokenCacheDictionary.Keys.FirstOrDefault().Authority);
-        }*/
+
+            // all mocks are consumed
+            Assert.AreEqual(0, HttpMessageHandlerFactory.MockHandlersCount());
+        }
+
+        [TestMethod]
+        [Description("Instance discovery call is made because authority was not already in the instance cache")]
+        public async Task AuthorityNotInInstanceCache_InstanceDiscoverCallMadeTestAsync()
```

Comment:
this test case should behave just like the one above where we get the tokens as expected. Only caveat is that InstanceDiscovery makes another call because DE cloud authority is not in the cache.

Reply:
I removed extra code that wasn't needed

Reply:
this test looks exactly like the one above. what is the goal of this test then? I think we have a misunderstanding here. can you please outline what this test is about and what are the network calls that will be made and why?